### ISSUE TITLE
No freedom in key at emotion/cache

### ISSUE THREAD

**Issue by usagi-f** [emotion-js/emotion#1542]
**The problem**

https://github.com/emotion-js/emotion/blob/master/packages/cache/src/index.js#L59

This regex is too strict for me. How the reason to force kebab-case?
I think to this spec is effect to architecture a little.

**Proposed solution**

For example, permit the following.

- A-Z
- underscore
- (Major) symbol

**Comment by Andarist** [comment]
How is it limiting you? How is it important beside aesthetical reasons?

**Comment by usagi-f** [comment]
Now, I'm thinking a mechanism that combines multiple env.

I know the unique key of cache is necessary.
So, we are considering adding a prefix or suffix to it.

However, there are few available patterns within the currently permitted range. also difficult to parse.

My thinking patterns; as example

hard to parse : `prefix-foo-bar-suffix`
error: `prefix_foo-bar_suffix`
error: `prefix#fooBar`

It's just an idea. How do you think about this?

**Comment by usagi-f** [comment]
It’s also good.

**Comment by Andarist** [comment]
Seems like the proposed workaround is "good enough" and there is no strong argument in favor of relaxing the `key` constraint. I hope this is not the problem for you - if it still is then please answer back with reasoning why.

**Comment by anilanar** [comment]
We have a unique identifier with a version number that prevents us from using the key. Why not allow valid css characters?
